### PR TITLE
[B2BP-305] - Add cloudfront function Terraform for add html extension

### PR DESCRIPTION
**Pull request by maurodandrea**
Cloudfront function for add html extension in static site request
Depends on #121 #124

#### List of Changes
Modify Terraform file

#### Motivation and Context
Cloudfront function for add html extension in static site request

#### Types of changes

- [X] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [X] My change not requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by datalek — CHANGES_REQUESTED** [review]
You need to add the build step to the [infrastructure deployment workflow](https://github.com/pagopa/b2b-portals/blob/main/.github/workflows/deploy_infra.yml#L40-L58); otherwise, the file to be uploaded won't be present.

Take a look at [this](https://github.com/pagopa/developer-portal/blob/46859b3/.github/workflows/deploy_infra.yml#L51-L52) as a reference, also on the workflow, set the parameter for publishing to true, as in [this line](https://github.com/pagopa/developer-portal/blob/46859b3/.github/workflows/deploy_infra.yml#L70-L71).

File: .infrastructure/60_cdn.tf [hunk 1]
```diff
@@ -2,6 +2,15 @@ resource "aws_cloudfront_origin_access_identity" "main" {
   comment = "Identity to access S3 bucket."
 }
 
+## Function to manipulate the request
+resource "aws_cloudfront_function" "website_viewer_request_handler" {
+  name    = "website-viewer-request-handler"
+  runtime = "cloudfront-js-1.0"
+  # publish this version only if the env is true
+  publish = var.publish_cloudfront_functions
+  code    = file("${path.module}/../apps/cloudfront-functions/src/viewer-request-handler.js")
```
Comment (datalek) [inline .infrastructure/60_cdn.tf:11]: I think the path is incorrect; it should be in `dist` instead of `src`

**Review by datalek — CHANGES_REQUESTED** [review]

File: .infrastructure/60_cdn.tf (hunk 1 above)
Comment (maurodandrea) [reply .infrastructure/60_cdn.tf:11]: correct

File: .infrastructure/60_cdn.tf [hunk 2]
```diff
@@ -2,6 +2,15 @@ resource "aws_cloudfront_origin_access_identity" "main" {
   comment = "Identity to access S3 bucket."
 }
 
+## Function to manipulate the request
+resource "aws_cloudfront_function" "website_viewer_request_handler" {
+  name    = "website-viewer-request-handler"
+  runtime = "cloudfront-js-1.0"
```
Comment (kin0992) [inline .infrastructure/60_cdn.tf:8]: I think we could try with `cloudfront-js-2.0`, what do you think?

Reference: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudfront_function#argument-reference

Comment (maurodandrea) [reply .infrastructure/60_cdn.tf]: upgraded to cloudfront-js-2.0 by new commit

**Review by datalek — APPROVED** [review]
LGTM

**Pull request merged**